coordinator/src/main/scala/filodb.coordinator/queryplanner/MultiPartitionPlanner.scala

```diff
@@ -22,15 +22,34 @@ import filodb.query.LogicalPlan._
 import filodb.query.exec._
 
 //scalastyle:off file.size.limit
+case class PartitionDetails(partitionName: String, httpEndPoint: String,
+                            grpcEndPoint: Option[String], proportion: Float)
+trait PartitionAssignmentTrait {
+  val proportionMap: Map[String, PartitionDetails]
+  val timeRange: TimeRange
+}
 
 case class PartitionAssignment(partitionName: String, httpEndPoint: String, timeRange: TimeRange,
-                               grpcEndPoint: Option[String] = None)
+                               grpcEndPoint: Option[String] = None) extends PartitionAssignmentTrait {
+  val proportionMap: Map[String, PartitionDetails] =
+    Map(partitionName -> PartitionDetails(partitionName, httpEndPoint, grpcEndPoint, 1.0f))
+}
+case class PartitionAssignmentV2(proportionMap: Map[String, PartitionDetails],
```

Comment:
should we add an assertion in `PartitionAssignmentV2` such that all the `proportion` in `proportionMap` will sum up to 1.0f ?